db/migrate/20181105212016_duplicate_deleted_at_state_changed_at.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class DuplicateDeletedAtStateChangedAt < ActiveRecord::Migration
+  def up
+    begin
+      Account.transaction do
+        Account.scheduled_for_deletion.where.has{ deleted_at != nil }.find_each do |account|
+          account.update_column(:state_changed_at, account.deleted_at) or raise ActiveRecord::Error
+        end
+      end
+    rescue ActiveRecord::Error
```

Comment:
You do not have to raise and rescue
If there is an error in a transaction it will rollback immediately